Discriminator Static Methods Not Properly Typed

### Prerequisites

- [x] I have written a descriptive issue title
- [x] I have searched existing issues to ensure the bug has not already been reported


### Mongoose version

8.18.0

### Node.js version

22.16.0

### MongoDB server version

6.8.0

### Typescript version (if applicable)

5.8.3

### Description

Discriminator models do not inherit static methods from their base models in TypeScript, despite working correctly at runtime. This causes TypeScript compilation errors when trying to access base model static methods on discriminator models, even though the functionality works perfectly at runtime.

The root cause is that the Model interface lacks a `TStaticMethods` generic parameter, so the discriminator method's return type can only include discriminator-specific static methods, not the base model's static methods.

**Current Model Interface Issue:**
```typescript
// Missing TStaticMethods generic parameter
export interface Model<
  TRawDocType,
  TQueryHelpers = {},
  TInstanceMethods = {},
  TVirtuals = {},
  THydratedDocumentType = HydratedDocument<...>,
  TSchema = any
>
```

**Discriminator Return Type Issue:**
```typescript
// Only includes discriminator static methods, missing base static methods
discriminator<TDiscriminatorSchema extends Schema<any, any>>(
  // ...
): Model<...> & ObtainSchemaGeneric<TDiscriminatorSchema, 'TStaticMethods'>;
```

This affects all applications using discriminators with custom static methods, causing TypeScript users to lose type safety and IntelliSense support despite correct runtime behavior.


### Steps to Reproduce

```typescript
// reproduction-script.ts
import { Schema, model } from 'mongoose';

// 1. Base model with custom static method
const baseSchema = new Schema({ name: String });
baseSchema.statics.findByName = function(name: string) {
  return this.findOne({ name });
};
const BaseModel = model('Base', baseSchema);

// 2. Create discriminator model
const discriminatorSchema = new Schema({ extra: String });
const DiscriminatorModel = BaseModel.discriminator('Discriminator', discriminatorSchema);

// 3. Try to use base static method - THIS CAUSES TYPESCRIPT ERROR
DiscriminatorModel.findByName('test');
//                 ^^^^^^^^^^
// TypeScript Error: Property 'findByName' does not exist on type 'Model<...>'

// 4. Verify it works at runtime (suppress TypeScript errors to test runtime behavior)
// @ts-ignore - Suppress error to prove runtime functionality
console.log('Runtime inheritance works:', typeof DiscriminatorModel.findByName); // "function"
```

**TypeScript compilation result:**
```bash
$ npx tsc reproduction-script.ts
error TS2339: Property 'findByName' does not exist on type 'Model<...>'
```

**Runtime execution result:**
```bash
$ npx ts-node reproduction-script.ts
Runtime inheritance works: function
```

### Expected Behavior

**Fixed version that should work:**
```typescript
// expected-behavior.ts - How it should work after the fix
import { Schema, model } from 'mongoose';

// Base model with custom static method
const baseSchema = new Schema({ name: String });
baseSchema.statics.findByName = function(name: string) {
  return this.findOne({ name });
};
const BaseModel = model('Base', baseSchema);

// Discriminator model inheriting base static methods
const discriminatorSchema = new Schema({ extra: String });
const DiscriminatorModel = BaseModel.discriminator('Discriminator', discriminatorSchema);

// ✅ Should work without TypeScript errors after fix
DiscriminatorModel.findByName('test'); // Proper typing inherited from base

console.log('TypeScript compilation should succeed!');
```

--- a/test/types/discriminator.test.ts
+++ b/test/types/discriminator.test.ts
@@ -116,3 +116,28 @@ function gh15535() {
   expectType<string | null | undefined>(doc.field3);
   expectType<string | null | undefined>(doc.getField3());
 }
+
+async function gh15600() {
+  // Base model with custom static method
+  const baseSchema = new Schema(
+    { name: String },
+    {
+      statics: {
+        findByName(name: string) {
+          return this.findOne({ name });
+        }
+      }
+    }
+  );
+  const BaseModel = model('Base', baseSchema);
+
+  const baseRes = await BaseModel.findByName('test');
+  expectType<string | null | undefined>(baseRes!.name);
+
+  // Discriminator model inheriting base static methods
+  const discriminatorSchema = new Schema({ extra: String });
+  const DiscriminatorModel = BaseModel.discriminator('Discriminator', discriminatorSchema);
+
+  const res = await DiscriminatorModel.findByName('test');
+  expectType<string | null | undefined>(res!.name);
+}

--- a/types/models.d.ts
+++ b/types/models.d.ts
@@ -435,7 +435,7 @@ declare module 'mongoose' {
       TQueryHelpers & ObtainSchemaGeneric<TDiscriminatorSchema, 'TQueryHelpers'>,
       TInstanceMethods & ObtainSchemaGeneric<TDiscriminatorSchema, 'TInstanceMethods'>,
       TVirtuals & ObtainSchemaGeneric<TDiscriminatorSchema, 'TVirtuals'>
-    > & ObtainSchemaGeneric<TDiscriminatorSchema, 'TStaticMethods'>;
+    > & ObtainSchemaGeneric<TSchema, 'TStaticMethods'> & ObtainSchemaGeneric<TDiscriminatorSchema, 'TStaticMethods'>;
     discriminator<D>(
       name: string | number,
       schema: Schema,

